packages/relay/src/lib/eth.ts

```diff
@@ -1578,46 +1583,24 @@ export class EthImpl implements Eth {
         );
 
       txSubmitted = true;
-      fileId = sendRawTransactionResult!.fileId;
-
-      // Wait for the record from the execution.
-      const txId = sendRawTransactionResult!.txResponse.transactionId.toString();
-      const formattedId = formatTransactionIdWithoutQueryParams(txId);
-
-      // handle formattedId being null
-      if (!formattedId) {
-        throw predefined.INTERNAL_ERROR(`Invalid transactionID: ${txId}`);
-      }
-
-      const contractResult = await this.mirrorNodeClient.repeatedRequest(
-        this.mirrorNodeClient.getContractResult.name,
-        [formattedId, requestDetails],
-        this.mirrorNodeClient.getMirrorNodeRequestRetryCount(),
-        requestDetails,
-      );
-
-      if (!contractResult) {
-        this.logger.warn(`${requestIdPrefix} No record retrieved`);
-        throw predefined.INTERNAL_ERROR(`No matching record found for transaction id ${txId}`);
-      }
-
-      if (contractResult.hash == null) {
-        this.logger.error(
-          `${requestIdPrefix} The ethereumHash can never be null for an ethereum transaction, and yet it was!!`,
+      fileId = sendRawTransactionResult.fileId;
+      submittedTransactionId = sendRawTransactionResult.txResponse.transactionId?.toString();
+      formattedTransactionId = formatTransactionIdWithoutQueryParams(submittedTransactionId);
```

Comment:
I also considered using the same submitted transactionID and reusing it outside the try-catch block as suggested in [your comment down here](https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229#discussion_r1833946368). However, there's a check for `Transaction successfully submitted but returned invalid transactionID` that is only relevant within the scope of this try block. 

Technically, I don’t believe it's possible for a successfully submitted transaction to return an invalid transactionID, but that doesn’t mean we shouldn’t account for potential edge cases. Additionally, this check is part of the original implementation, and I’d prefer not to remove it within the scope of this PR.

Reply:
What I meant is that we can define only one field - `transactionId` and use only that field throughout the method. Since `formattedTransactionId` is used only for the sake of calling the mirror node, I don't see a reason we would define it as a global variable in the method. 

Instead, what I suggested is to format it right before we send the request to the mirror node as shown in https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229#discussion_r1833946368 and have it as a local variable in that part of the logic, instead of maintaining it globally on the method level (I hope that makes sense to you)

Reply:
Yeah that makes total sense! I initially kept `formattedTransactionId` because I was relying on `formatTransactionIdWithoutQueryParams()` to verify the transaction’s validity by checking `transactionID` against `constants.TRANSACTION_ID_REGEX`. That’s why I reused the formatted ID globally. However, after looking into `formatTransactionIdWithoutQueryParams()`, it seems I can simply check `transactionID` directly within the try block without needing to format it.

Thanks for making me to reconsider this you’re absolutely right! There’s no need to maintain `formattedTransactionId` globally.

Reply:
Pushed in an update in this commit https://github.com/hashgraph/hedera-json-rpc-relay/pull/3229/commits/e56d4a5e184b28021f9c0e43b0bd69b9504d7476. 

Nice job pointing it out 🙌🏼